### ISSUE TITLE
bootstrap pod does not have a way to set securityContext

### ISSUE THREAD

**Issue by jbnjohnathan** [Opster/opensearch-k8s-operator#606]
While securityContext can be set on the `opensearch` and `opensearch-dashboard` pods, it cannot be set on the bootstrap pod.

Operator version: `2.4.0`

I set the securityContext in `spec.general.securityContext` in the values file.

```
  opensearchCluster:
    initHelper:
      image: public.ecr.aws/opsterio/busybox
      version: 1.27.2-buildx
  #  bootstrap:
  #    Configure settings for the bootstrap pod
    general:
      image: docker.io/opensearchproject/opensearch
      version: *opensearch-version
      httpPort: "9200"
      serviceName: "opensearch-http"
      drainDataNodes: true
      setVMMaxMapCount: true
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop:
            - ALL
        readOnlyRootFilesystem: true
        runAsNonRoot: true
        seccompProfile:
          type: RuntimeDefault
```

```
"error":"failed to create resource: creating resource failed: pods \"opensearch-bootstrap-0\" is forbidden: unable to validate against any security context constraint: [provider \"anyuid\": Forbidden: not usable by user or serviceaccount, spec.initContainers[0].securityContext.privileged: Invalid value: true: Privileged containers are not allowed, provider \"restricted\": Forbidden: not usable by user or serviceaccount, provider \"nonroot-v2\": Forbidden: not usable by user or serviceaccount, provider \"nonroot\": Forbidden: not usable by user or serviceaccount, provider \"hostmount-anyuid\": Forbidden: not usable by user or serviceaccount, provider \"machine-api-termination-handler\": Forbidden: not usable by user or serviceaccount, provider \"hostnetwork-v2\": Forbidden: not usable by user or serviceaccount, provider \"hostnetwork\": Forbidden: not usable by user or serviceaccount, provider \"hostaccess\": Forbidden: not usable by user or serviceaccount, provider \"node-exporter\": Forbidden: not usable by user or serviceaccount, provider \"privileged\": Forbidden: not usable by user or serviceaccount]","errorVerbose":"pods \"opensearch-bootstrap-0\" is forbidden: unable to validate against any security context constraint: [provider \"anyuid\": Forbidden: not usable by user or serviceaccount, spec.initContainers[0].securityContext.privileged: Invalid value: true: Privileged containers are not allowed, provider \"restricted\": Forbidden: not usable by user or serviceaccount, provider \"nonroot-v2\": Forbidden: not usable by user or serviceaccount, provider \"nonroot\": Forbidden: not usable by user or serviceaccount, provider \"hostmount-anyuid\": Forbidden: not usable by user or serviceaccount, provider \"machine-api-termination-handler\": Forbidden: not usable by user or serviceaccount, provider \"hostnetwork-v2\": Forbidden: not usable by user or serviceaccount, provider \"hostnetwork\": Forbidden: not usable by user or serviceaccount, provider \"hostaccess\": Forbidden: not usable by user or serviceaccount, provider \"node-exporter\": Forbidden: not usable by user or serviceaccount, provider \"privileged\": Forbidden: not usable by user or serviceaccount]\ncreating resource failed
```

I mentioned this in another issue, however that issue was not created for this error, so I am creating this separate issue about it.
Previous issue: https://github.com/Opster/opensearch-k8s-operator/issues/598

**Comment by jbnjohnathan** [comment]
FYI The discussion about this issue continues in #598

**Comment by swoehrl-mw** [comment]
Closing as solved via discussion in #598.
